sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -295,13 +295,14 @@ public final class StorageBlobClient: PipelineClient {
     /// - Parameters:
     ///   - options: A `ListContainersOptions` object to control the list operation.
     ///   - completionHandler: A completion handler that receives a `PagedCollection` of `ContainerItem` objects on success.
+    @discardableResult
     public func listContainers(
         withOptions options: ListContainersOptions? = nil,
         completionHandler: @escaping HTTPResultHandler<PagedCollection<ContainerItem>>
-    ) {
+    ) -> AzureTask {
         // Construct URL
         let urlTemplate = ""
-        guard let url = self.url(forTemplate: urlTemplate) else { return }
+        guard let url = self.url(forTemplate: urlTemplate) else { return AzureTask.for(error: "Precondition failure") }
```

Comment:
General question - when we error out inside the body of the public API method like this, should we also be calling the completionHandler with the error?

Reply:
That's a good question. Philosophically I would argue no since it is an HTTPResultHandler, and if it errors out as a precondition, there's really no HTTP result to speak of. On the other hand, it would be inconvenient/annoying to need to check for an error both in the completion handler and in the task object. Should probably be a discussion with @adrianhall  and the Android folks as well.

Reply:
In discussing with Adrian, he mentioned it would make sense to call the callback with the failure immediately. This makes more sense if we decide to not go with the approach of returning an AzureTask.